torch/testing/_internal/common_methods_invocations.py

```diff
@@ -5120,6 +5121,25 @@ def c(t):
     inputs.append(SampleInput([c(H)], args=("i...->...",)))
     inputs.append(SampleInput([c(C), c(x)], args=('...ik, ...j -> ij',)))
 
+    # Test optimize kwarg
+    z = make_tensor((3, 2), device=device, dtype=dtype, requires_grad=requires_grad)
+    inputs.append(SampleInput([c(A), z], args=('ij,jk',), kwargs={'optimize': [(0, 1)]}))
+    inputs.append(SampleInput([c(A), c(D), c(C), c(E)],
+                  args=('jk,ikl,ijk,lm->mk',),
+                  kwargs={'optimize': [(0, 2), (0, 1), (0, 1)]}))
+    if TEST_OPT_EINSUM:
+        inputs.append(SampleInput([c(A), c(D), c(C), c(E)],
```

Comment:
fyi these SampleInputs could be updated to just yield each sample, and thanks to @peterbell10 the sample inputs can now be written "naturally" as regular inputs without dealing with the `args` or `kwargs` params, like:

```
yield SampleInput([c(A), c(D), c(C), c(E)], ('jk,ikl,ijk,lm->mk',), optimize='auto')
```

Reply:
oh that is p cool!